engine/access/rpc/engine.go

```diff
@@ -324,43 +340,54 @@ func (e *Engine) serveSecureGRPC() {
 	e.addrLock.Unlock()
 
 	e.log.Debug().Str("secure_grpc_address", e.secureGrpcAddress.String()).Msg("listening on port")
+	ready()
 
 	err = e.secureGrpcServer.Serve(l) // blocking call
 	if err != nil {
-		e.log.Fatal().Err(err).Msg("fatal error in secure grpc server")
+		e.log.Err(err).Msg("fatal error in secure grpc server")
+		ctx.Throw(err)
 	}
 }
 
-// serveGRPCWebProxy starts the gRPC web proxy server
-func (e *Engine) serveGRPCWebProxy() {
+// serveGRPCWebProxyWorker is a worker routine which starts the gRPC web proxy server.
+func (e *Engine) serveGRPCWebProxyWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
 	log := e.log.With().Str("http_proxy_address", e.config.HTTPListenAddr).Logger()
-
 	log.Info().Msg("starting http proxy server on address")
 
-	err := e.httpServer.ListenAndServe()
+	err := e.httpServer.ListenAndServe() // blocking call
```

Comment:
I recommend splitting into:
- listen
- ready
- go serve

like in serveSecureGRPCWorker

Reply:
Good idea, done in https://github.com/onflow/flow-go/pull/4313/commits/7b574e38173891804a84a240d96cdb3a5051b849